controllers/solrbackup_controller.go

```diff
@@ -253,7 +283,21 @@ func persistSolrCloudBackups(r *SolrBackupReconciler, backup *solrv1beta1.SolrBa
 	}
 	now := metav1.Now()
 
-	persistenceJob := util.GenerateBackupPersistenceJobForCloud(backup, solrCloud)
+	backupRepository, found := util.GetBackupRepositoryByName(solrCloud.Spec.StorageOptions.BackupRestoreOptions, backup.Spec.RepositoryName)
+	if ! found {
+		err = fmt.Errorf("Unable to find backup repository to use for backup [%s] (which specified the repository" +
+			"[%s]).  solrcloud must define a repository matching that name (or have only 1 repository defined).",
+			backup.Name, backup.Spec.RepositoryName)
+		return err
+	}
+
+	managedBackupRepository, ok := backupRepository.(util.ManagedBackupRepository)
+	if ! ok {
```

Comment:
I'd rather the persistence stuff go in this `if ok {`, otherwise it's a little hard to follow.